Compiler warning in embedded mode in v1.11.3

I noticed this compiler warning this morning when compiling Open MPI master (which embeds hwloc 1.11.3):

```
topology-x86.c: In function ‘hwloc_x86_discover’:
topology-x86.c:1094:5: warning: "HAVE_DECL_RUNNING_ON_VALGRIND" is not defined [-Wundef]
 #if HAVE_DECL_RUNNING_ON_VALGRIND
     ^
```

--- a/config/hwloc.m4
+++ b/config/hwloc.m4
@@ -673,6 +673,8 @@ EOF])
     AS_IF([test "$hwloc_mode" != "embedded"],
         [AC_CHECK_HEADERS([valgrind/valgrind.h])
          AC_CHECK_DECLS([RUNNING_ON_VALGRIND],,[:],[[#include <valgrind/valgrind.h>]])
+	],[
+	 AC_DEFINE([HAVE_DECL_RUNNING_ON_VALGRIND], [0], [Embedded mode; just assume we do not have Valgrind support])
 	])
 
     AC_CHECK_HEADERS([pthread_np.h])

